data/db_demo.yaml

```diff
@@ -12,6 +12,23 @@ user:
     roles:
       - View only
 
+streams:
+  - name: ZTF Public
+    altdata:
+      collection: ZTF_alerts
```

Comment:
What if I make a group that filters on streams 2 and 4 only. Will I need to make a new Stream object for 2+4 in addition to these three and grant it manually to each user and group? What if I had 2+1+4? Will I have to grant that to each user manually for them to be able to see a stream even if they have [1], and [2,4]? 

If so, I think that argues for having each stream map 1-to-1 to each PID, and granting those permissions individually to each user.

Reply:
I feel that this is a valid scenario abstractly speaking, but an artificial one in practice. Concretely, what would this 2+4 mean? In practice, for F we will have ztf_public, ztf_public+partnership, ztf_public+partnership+caltech, and gattini (and in the future, LSST, and potentially other surveys). So I do not quite understand what kinds of combinations you are talking about (and in general, this all PID stuff essentially only applies to ZTF). However, the implementation here in SP is quite general and none of the `altdata` stuff is required.

Next, these streams really point at a sink on the K side (even though eventually, SP may get its own stream listeners, but it is a completely different story), this is why I "aggregate" them this way, otherwise things will be nice, abstract, and clean on the SP side, but that would add unnecessary complexity on the K side of F.

Finally, yes, when a new stream is added to the system, a SU must add it individual users first, that's the scheme we converged on. Only when/if all users of a group have access to a particular (new) stream, it can be added to the group.